lib/Service/FreePrompt/FreePromptService.php

```diff
@@ -26,18 +27,17 @@ public function __construct(
 		private IManager $textProcessingManager,
 		private ?string $userId,
 		private PromptMapper $promptMapper,
-		private IL10N $l10n
+		private IL10N $l10n,
+		private TaskMapper $taskMapper,
 	) {
 	}
 
 	/*
 	 * @param string $prompt
-	 * @param int $nResults
-	 * @param bool $showPrompt
 	 * @return string
 	 * @throws Exception
 	 */
-	public function processPrompt(string $prompt, int $nResults): string {
+	public function processPrompt(string $prompt): string {
```

Comment:
Seems like this is duplicating the text processing task logic in AssistantService, no?

Reply:
Instead of having process and schedule methods all over the place, I could imagine we could make use of the visitor pattern with a RunVisitor, a ScheduleVisitor, a RunOrScheduleVisitor and perhaps a NotifyVisitor or similar that each visit a Task

Reply:
Yeah, I absolutely agree! I guess the question is if such refactoring should be a part of this pr?

Reply:
Yeah, let's not make it part of this PR